sw/host/opentitantool/src/backend/mod.rs

```diff
@@ -15,8 +17,8 @@ pub mod verilator;
 
 #[derive(Debug, StructOpt)]
 pub struct BackendOpts {
-    #[structopt(long, default_value, help = "Name of the debug interface")]
-    interface: String,
+    #[structopt(long, help = "Name of the debug interface")]
+    interface: Option<String>,
```

Comment:
Why turn this into `Option<String>`?

There are and will be commands which do not require a transport, which is the purpose that `EmptyTransport` serves (since all `CommandDispatch` implementors are required to take `Transport`, or `TransportWrapper` now).  The `Capabilities` struct allows commands to check their transports before initializing anything so they can return a reasonable error message (ie: if you try to `spi read` with an empty transport).

I think leaving this with `""` as the default will also make creating the `TransportWrapper` easier.  See my comments below.

Reply:
I had in mind a number of use  cases:
Invoking opentitantool with a set of configuration files, one of which specifies interface, the command line does not.
Invoking opentitantool with a set of configuration files, one of which specifies interface type, the command line specifying USB serial number.
Invoking opentitantool with no configuration files, specifying interface on the command line.
Invoking opentitantool with a set of configuration files, none of which mentions interface, specifying interface on the command line. 

I thought the cleanest way was to represent all of the interface specifications as Option<String>, (both from command line and configuration files), with the constraint that exactly one of them must be Some(_), the others None.  Since there is no real use for Some("") in my scheme, we could have done all the same with plain String, and the convention that the empty string means absence of any specification.